napi-inl.h

```diff
@@ -442,7 +442,7 @@ inline Value Env::RunScript(String script) {
 
 #if NAPI_VERSION > 5
 template <typename T, Env::Finalizer<T> fini>
-inline void Env::SetInstanceData(T* data) {
+inline void Env::SetInstanceData(T* data) const {
```

Comment:
This does conceptually modify the environment, because it attaches instance data to it. I know it doesn't modify the state of the `Napi::Env` object, but it does modify the state of `napi_env`.

Reply:
If attaching instance data to the environment is conceptually modifying the Env object, then this modification shouldn't affect copies of the object. But copies of Env share the same environment, similarly to how copies of shared_ptr point to the same object.